fpgalib/adc.py

```diff
@@ -441,9 +439,10 @@ def runAverage(self, filterFunc, filterStretchLen, filterStretchAt,
         @inlineCallbacks
         def func():
             # build registry packet
-            regs = self.regRun(self, self.RUN_MODE_AVERAGE_AUTO, 1, filterFunc,
-                filterStretchLen, filterStretchAt, demods)
-            
+            regs = self.regRun(self, mode=self.RUN_MODE_AVERAGE_AUTO, 
```

Comment:
I think I was being a bit overdramatic about classmethods. We should just remove the first `self` argument to the method. If you can do that here while you're touching these lines, @patzinak, that'd be great.